src/libraries/System.Text.Json/gen/JsonSourceGenerator.cs

```diff
@@ -15,22 +15,21 @@ namespace System.Text.Json.SourceGeneration
     /// Generates source code to optimize serialization and deserialization with JsonSerializer.
     /// </summary>
     [Generator]
-    public sealed class JsonSourceGenerator : ISourceGenerator
+    public sealed partial class JsonSourceGenerator : ISourceGenerator
     {
-        private JsonSourceGeneratorHelper? _helper;
-
         /// <summary>
         /// Helper for unit tests.
         /// </summary>
-        public Dictionary<string, Type>? SerializableTypes => _helper.GetSerializableTypes();
+        public Dictionary<string, Type>? SerializableTypes => _rootTypes?.ToDictionary(p => p.Key, p => p.Value.Type);
```

Comment:
This is a property, but it calls a Linq method, which will create a new Dictionary every time. This can be a perf trap, since properties should have field-access like performance.

Reply:
Changed this to be a method. The tests that use it have been updated to cache the return value.